lib/js/src/manager/screen/choiceset/_CheckChoiceVrOptionalOperation.js

```diff
@@ -0,0 +1,127 @@
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import { _Task } from '../../_Task';
+import { DeleteInteractionChoiceSet } from '../../../rpc/messages/DeleteInteractionChoiceSet.js';
+import { CreateInteractionChoiceSet } from '../../../rpc/messages/CreateInteractionChoiceSet.js';
+import { Choice } from '../../../rpc/structs/Choice.js';
+
+class _CheckChoiceVrOptionalOperation extends _Task {
+    /**
+     * Initializes an instance of _CheckChoiceVrOptionalOperation
+     * @class
+     * @private
+     * @param {_LifecycleManager} lifecycleManager - A _LifecycleManager instance
+     * @param {_CheckChoiceVrOptionalInterface} checkChoiceVrOptionalInterface - A callback function for operation updates
+     */
+    constructor (lifecycleManager, checkChoiceVrOptionalInterface = null) {
+        super('CheckChoiceVrOptionalOperation');
+        this._lifecycleManager = lifecycleManager;
+        this._checkChoiceVrOptionalInterface = checkChoiceVrOptionalInterface;
+        this._isVrOptional = null;
+    }
+
+    /**
+     * The method that causes the task to run.
+     * @param {_Task} task - The task instance
+     * @returns {Promise}
+     */
+    async onExecute (task) {
+        let deletePromise;
+
+        const noVrResponse = await this._lifecycleManager.sendRpcResolve(this._createTestChoiceSet(false));
+        if (noVrResponse.getSuccess()) {
+            this._isVrOptional = true;
+            deletePromise = await this._lifecycleManager.sendRpcResolve(this._deleteTestChoiceSet());
+        } else {
+            // head unit doesn't support choices with no VR
+            this._isVrOptional = false;
+            const vrResponse = await this._lifecycleManager.sendRpcResolve(this._createTestChoiceSet(true));
+            if (vrResponse.getSuccess()) {
+                console.warn('CheckChoiceVROptionalOperation: Connected head unit does not support choice cells without voice commands. Cells without voice will be sent with placeholder voices from now on.');
+                deletePromise = await this._lifecycleManager.sendRpcResolve(this._deleteTestChoiceSet());
+            } else {
+                console.error('CheckChoiceVROptionalOperation: Connected head unit has rejected all choice cells. Choice manager disabled');
+                if (this._checkChoiceVrOptionalInterface !== null) {
+                    this._checkChoiceVrOptionalInterface.onError(vrResponse.getInfo());
+                    this.onFinished();
+                    return;
+                }
+            }
+        }
+
+        const deleteResponse = await deletePromise;
```

Comment:
You're using await here but that was already done when deletePromise was initialized earlier.